src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -350,6 +350,7 @@ protected TypeDesc GetDefaultNativeType(NativeType nativeType)
                         return GetDefaultNativeType(NativeType.SysInt);
 
                 case MarshallerKind.UnicodeString:
+                    return GetDefaultNativeType(NativeType.SysInt); // We pin UniCodeString, so native type will be SysInt
```

Comment:
As a follow up cleanup, could you please add comment to GetDefaultNativeType about it does? Looking at how it is used, it seems to be mixing multiple different concerns together.